perception/autoware_traffic_light_arbiter/src/signal_match_validator.cpp

```diff
@@ -168,9 +169,16 @@ TrafficSignal get_highest_confidence_signal(
     return *perception_signal;
   }
 
-  // Gives priority to the external signal if external_priority is true
-  if (external_priority) {
-    return *external_signal;
+  // Apply source priority
+  switch (source_priority) {
+    case autoware::traffic_light::SourcePriority::EXTERNAL:
+      return *external_signal;
+    case autoware::traffic_light::SourcePriority::PERCEPTION:
+      return *perception_signal;
+    case autoware::traffic_light::SourcePriority::CONFIDENCE:
+    default:
```

Comment:
[imho]
Using a default case here makes it harder to detect newly added SourcePriority values, since they would silently fall through to the CONFIDENCE branch. It’s safer to remove the default and handle each enum explicitly, so that future extensions trigger a compile-time error.